model/src/DA/Finance/Types.daml

```diff
@@ -63,3 +63,19 @@ data MasterAgreement = MasterAgreement
     party2 : Party
       -- ^ Allows to specify choices of the second counterparty.
   deriving (Eq, Show)
+
+-- | A record for templates which can be locked. Also includes a set of
+--  parties which may unlock a locked template.
+data Lock = Lock
+  with
+    by : Party
```

Comment:
Are there cases where we need to know explicitly who put a particular lock in place? Similar to the below timestamp, the information is implicitly present as the submitter of the transaction placing the lock. So unless there's a specific use case where we do need this explicitly, I'd suggest to leave it out for now.

Reply:
As per timestamp field, I saw this field as useful information to have for both display and analysis purposes. Ie., if you're a client and you're looking at your account overview, it would be useful to know who locked your deposits.